scipy/sparse/csgraph/_disjoint_set.py

```diff
@@ -0,0 +1,137 @@
+"""
+Disjoint set data structure
+"""
+
+
+import numpy as np
+
+
+class DisjointSet:
+    """ Disjoint set data structure for incremental connectivity queries.
+
+    .. versionadded: 1.6.0
+
+    Attributes
+    ----------
+    n_nodes : int
+        The number of nodes in the set.
+    n_components : int
+        The number of components/subsets.
+
+    Methods
+    -------
+    union
+    find
+
+    Notes
+    -----
+    This class implements the disjoint set [1]_, also known as the *union-find*
+    data structure. The *find* method implements the *path compression*
+    variant. The *union* method implements the *union by size* variant.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Disjoint-set_data_structure
+
+    See Also
+    --------
+    connected_components : Analyze the connected components of a sparse graph.
+
+    Examples
+    --------
+    >>> from scipy.sparse.csgraph import DisjointSet
+
+    Initialize a disjoint set with 4 nodes:
+
+    >>> dis = DisjointSet(4)
```

Comment:
The initializer signature seems to have changed along the way, so this one should result in an argument count mismatch.

Reply:
Good catch! Thanks.